src/lumigo_tracer/tracer.py

```diff
@@ -0,0 +1,163 @@
+import inspect
+import logging
+import builtins
+from functools import wraps
+
+from lumigo_tracer.auto_tag.auto_tag_event import AutoTagEvent
+from lumigo_tracer.lumigo_utils import (
+    config,
+    Configuration,
+    get_logger,
+    lumigo_safe_execute,
+    is_aws_environment,
+    is_kill_switch_on,
+)
+from lumigo_tracer.spans_container import SpansContainer, TimeoutMechanism
+from lumigo_tracer.wrappers import wrap
+
+CONTEXT_WRAPPED_BY_LUMIGO_KEY = "_wrapped_by_lumigo"
+
+
+def _is_context_already_wrapped(*args) -> bool:
+    """
+    This function is here in order to validate that we didn't already wrap this lambda
+        (using the sls plugin / auto instrumentation / etc.)
+    """
+    return len(args) >= 2 and hasattr(args[1], CONTEXT_WRAPPED_BY_LUMIGO_KEY)
+
+
+def _add_wrap_flag_to_context(*args):
+    """
+    This function is here in order to validate that we didn't already wrap this invocation
+        (using the sls plugin / auto instrumentation / etc.).
+    We are adding lumigo's flag to the context, and check it's value in _is_context_already_wrapped.
+    """
+    if len(args) >= 2:
+        with lumigo_safe_execute("wrap context"):
+            setattr(args[1], CONTEXT_WRAPPED_BY_LUMIGO_KEY, True)
+
+
+def _lumigo_tracer(func):
+    @wraps(func)
+    def lambda_wrapper(*args, **kwargs):
```

Comment:
Check this condition outside the lambda wrapper, and just return `func` if it switched off